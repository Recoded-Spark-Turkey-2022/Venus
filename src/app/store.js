// eslint-disable-next-line no-unused-vars
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import listingReducer from '../features/listings/listingSlice';

export const store = configureStore({
  reducer: {
    listing: listingReducer,
  },
  // eslint-disable-next-line no-shadow
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});
