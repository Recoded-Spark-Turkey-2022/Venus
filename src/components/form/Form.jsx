import './form.css';
import man from '../../assets/man.png';
import img from '../../assets/signup-vector.svg';

const Form = () => {
  return (
    <section className="h-[130vh] section-form overflow-hidden w-screen md:h-screen relative ">
      <img
        src={img}
        className="2xl:mt-16 absolute image-hero z-0 left-[-500px] md:left-[-200px] lg:left-[-300px] xl:left-[-400px] 2xl:left-[-600px] top-[0px]  w-[100%] h-[110%] "
        alt="circle-logo"
      />
      <div className="BigContainer">
        <div className="Container1">
          <img src={man} alt="p" className="manImg" />

          <button className="cam-btn" type="button">
            <ion-icon size="large" name="camera-outline" />
          </button>
          <h1 className="name">Hi I am Here</h1>
        </div>
        <div className="Container2">
          <form>
            <div className="name-sur">
              <div className="name-div">
                <h1>Name:</h1>
                <input type="text" name="name" className="nameform" />
              </div>
              <div className="sur-div">
                <h1>Surname:</h1>
                <input type="text" name="surname" className="surname" />
              </div>
            </div>
            <div className="Bio">
              <div className="Bio-div">
                <h1>Biography:</h1>
                <input type="text" name="Biography" className="Biography" />
              </div>
              <div className="Loc-div">
                <h1>Location:</h1>
                <input type="text" name="Location" className="Location" />
              </div>
            </div>
          </form>
        </div>
        <div className="Container4">
          <button type="submit" className="save-btn">
            SAVE
          </button>
          <button type="submit" className="cancel-btn">
            CANCEL
          </button>
        </div>
      </div>
    </section>
  );
};

export default Form;
