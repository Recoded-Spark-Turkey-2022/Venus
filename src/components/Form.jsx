import React from "react";
import './form.css'
import man from '../assets/man.png'
import img from '../assets/signup-vector.svg'



const Form = () => {
    return (
        <section className=' w-screen h-screen relative '>
          <img
           src={img}
           className="2xl:mt-16 absolute image-hero z-0 left-[-500px] md:left-[-200px] lg:left-[-300px] xl:left-[-400px] 2xl:left-[-600px] top-[0px]  w-[100%] h-[90vh] "
           alt="circle-logo"/>
          <div className="BigContainer">
           <div className="Container1">
           <img src={man} alt="p" className="manImg"/>
           
           <button className='cam-btn' type='button'>
            <ion-icon size='large' name="camera-outline" />
                   </button>
                   <h1 className="name">tasnim al hallak</h1>
                   
           </div> 
           <div className="Container2">awe</div>
           <div className="Container4">awe</div>
          </div>
          

         
      </section>

    )
     } 


 export default Form;