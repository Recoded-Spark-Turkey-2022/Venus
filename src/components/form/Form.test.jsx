import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../app/store';
import Form from './Form';

const props = {
    setOpen: 'test',
    onChange: 'test',
    file: 'test',
    image: 'test',
    setImage: 'test',
    setName: 'test',
    name: 'test'
  };

describe('Form', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(
            <Provider store={store}>
              <Form data={props} />
            </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });