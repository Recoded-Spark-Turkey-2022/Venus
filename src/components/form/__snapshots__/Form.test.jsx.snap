// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders correctly 1`] = `
<section
  className="h-[145vh] sm:h-[130vh] section-form overflow-hidden w-screen md:h-[120vh] relative "
>
  <img
    alt="circle-logo"
    className="2xl:mt-16 absolute image-hero z-0 left-[-500px] md:left-[-200px] lg:left-[-300px] xl:left-[-400px] 2xl:left-[-600px] top-[0px]  w-[100%] h-[110%] "
    src="signup-vector.svg"
  />
  <div
    className="BigContainer"
  >
    <div
      className="Container1"
    >
      <form
        className=" flex flex-col items-center justify-center mb-5"
        onSubmit={[Function]}
      >
        <div
          className="image-wrapper"
        >
          <img
            alt="profile"
            className="manImg"
            src="https://icon-library.com/images/no-image-icon/no-image-icon-0.jpg"
          />
          <label
            className="cam-btn"
            htmlFor="file"
          >
            <ion-icon
              name="camera-outline"
              size="large"
            />
          </label>
        </div>
        <input
          id="file"
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
        <button
          className="rounded-2xl py-1 px-2 "
          id="mediumBlue-button"
          type="submit"
        >
          Upload Avatar
        </button>
      </form>
      <h1
        className="name"
      >
        Hi I am 
         
      </h1>
    </div>
    <div
      className="Container2"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="name-sur"
        >
          <div
            className="name-div"
          >
            <h1>
              Name:
            </h1>
            <input
              className="nameform"
              max={18}
              maxLength={18}
              name="name"
              required={true}
              type="text"
            />
          </div>
          <div
            className="sur-div"
          >
            <h1>
              Surname:
            </h1>
            <input
              className="surname"
              max={15}
              name="surname"
              type="text"
            />
          </div>
        </div>
        <div
          className="Bio"
        >
          <div
            className="Bio-div"
          >
            <h1>
              Biography:
            </h1>
            <input
              className="Biography"
              min={5}
              name="Biography"
              required={true}
              type="text"
            />
          </div>
          <div
            className="Loc-div"
          >
            <h1>
              Location:
            </h1>
            <input
              className="Location"
              name="Location"
              required={true}
              type="text"
            />
          </div>
        </div>
        <div
          className="Container4"
        >
          <button
            className="save-btn"
            type="submit"
          >
            SAVE
          </button>
          <button
            className="cancel-btn"
            onClick={[Function]}
            type="button"
          >
            CANCEL
          </button>
        </div>
      </form>
    </div>
  </div>
  <div
    className="Toastify"
  />
</section>
`;
